### PR TITLE
[API][Checkout] Cover scenarios for checking promotions

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Review by AdamKasp — APPROVED** [review]

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php [hunk 1]
```diff
@@ -61,13 +71,94 @@ public function iShouldBeAbleToAccessThisOrderDetails(): void
...
 
     /**
-     * @Then /^my tax total should be ("[^"]+")$/
+     * @Then :promotionName should be applied to my order
      */
-    public function myTaxTotalShouldBe(int $taxTotal): void
+    public function shouldBeAppliedToMyOrder(string $promotionName): void
     {
-        $response = $this->client->show($this->sharedStorage->get('cart_token'));
+        Assert::notFalse(array_search($promotionName, array_column($this->getAdjustmentsForOrder(), 'label')));
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:78]: This extraction logic seems like a good candidate for placing it in separate private method

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php [hunk 2]
```diff
@@ -61,13 +71,94 @@ public function iShouldBeAbleToAccessThisOrderDetails(): void
...
+    }
+
+    /**
+     * @Then :promotionName should be applied to my order shipping
+     */
+    public function shouldBeAppliedToMyOrderShipping(string $promotionName): void
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:84]: Can we merge it with the previous step?

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php [hunk 3]
```diff
@@ -61,13 +71,94 @@ public function iShouldBeAbleToAccessThisOrderDetails(): void
...
+    {
+        $adjustments = $this->getAdjustmentsForOrder();
+        Assert::same(
+            $discount,
+            $adjustments[array_search($promotion->getName(), array_column($adjustments, 'label'))]['amount']
+        );
+    }
+
+    /**
+     * @Then /^the ("[^"]+" product) should have unit price discounted by ("\$\d+")$/
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:102]: Why didn't you use our price transformation here?

File: src/Sylius/Behat/Context/Api/Shop/OrderContext.php [hunk 4]
```diff
@@ -61,13 +71,94 @@ public function iShouldBeAbleToAccessThisOrderDetails(): void
...
+        $response = $this->client->subResourceIndex('adjustments', $this->sharedStorage->get('cart_token'));
+
+        return $this->responseChecker->getCollection($response);
+    }
+
+    private function getAdjustmentsForOrderItem(string $itemId): array
+    {
+        $response = $this->client->customAction(
+            sprintf('/new-api/orders/%s/items/%s/adjustments', $this->sharedStorage->get('cart_token'), $itemId),
+            HttpRequest::METHOD_GET);
```
Comment (lchrusciel) [inline src/Sylius/Behat/Context/Api/Shop/OrderContext.php:127]:
```suggestion
            HttpRequest::METHOD_GET)
        ;
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Adjustment.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        <collectionOperations />
+
+        <itemOperations>
+            <itemOperation name="get">
+                <attribute name="security">is_granted('ROLE_API_ACCESS')</attribute>
+            </itemOperation>
+        </itemOperations>
+
+        <subresourceOperations>
+            <subresourceOperation name="api_order_items_adjustments_get_subresource">
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Adjustment.xml:34]: What is the difference between these two operations?

File: src/Sylius/Bundle/ApiBundle/spec/DataProvider/OrderAdjustmentsSubresourceDataProviderSpec.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    function it_supports_only_order_adjustments_subresource_data_provider(): void
+    {
+        $this
+            ->supports(ProductInterface::class, Request::METHOD_GET)
+            ->shouldReturn(false)
+        ;
+
+        $context['subresource_identifiers'] = ['id' => 'TOKEN'];
```
Comment (lchrusciel) [inline src/Sylius/Bundle/ApiBundle/spec/DataProvider/OrderAdjustmentsSubresourceDataProviderSpec.php:38]: We should test the version with `items` as well

**Review by lchrusciel — APPROVED** [review]

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Adjustment.xml (hunk 5 above)
Comment (GSadee) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/Adjustment.xml:34]: Using the subresources to generate adjustments endpoints, for example to have endpoint: `/orders/{token}/items/{id}/adjustments`, we need to expose also ``/order-items/{id}/adjustments``. 

These are all adjustments endpoints that are currently exposed:
image

**Comment by lchrusciel** [comment]
Thank you, Grzegorz! :tada:

**Pull request merged**